python/lsst/pipe/tasks/dcrModel.py

```diff
@@ -0,0 +1,539 @@
+        Sub-region of the masked image to shift.
+        Shifts the entire image if None (Default).
+    useInverse : `bool`, optional
+        Use the reverse of ``dcr`` for the shift. Default: False
+
+    Returns
+    -------
+    `lsst.afw.image.maskedImageF`
+        A masked image, with the pixels within the bounding box shifted.
+
+    Deleted Parameters
+    ------------------
+    useFFT : `bool`, optional
+        Perform the convolution with an FFT?
+    """
+    padValue = afwImage.pixel.SinglePixelF(0., maskedImage.mask.getPlaneBitMask("NO_DATA"), 0)
+    if bbox is None:
+        bbox = maskedImage.getBBox()
+    shiftedImage = afwImage.MaskedImageF(bbox)
+    transform = makeTransform(AffineTransform((-1.0 if useInverse else 1.0)*dcr))
+    afwMath.warpImage(shiftedImage, maskedImage[bbox],
+                      transform, warpCtrl, padValue=padValue)
+    return shiftedImage
+
+
+def calculateDcr(visitInfo, wcs, filterInfo, dcrNumSubfilters):
+    """Calculate the shift in pixels of an exposure due to DCR.
+
+    Parameters
+    ----------
+    visitInfo : `lsst.afw.image.VisitInfo`
+        Metadata for the exposure.
+    wcs : `lsst.afw.geom.SkyWcs`
+        Coordinate system definition (wcs) for the exposure.
+    filterInfo : `lsst.afw.image.Filter`
+        The filter definition, set in the current instruments' obs package.
+    dcrNumSubfilters : `int`
+        Number of sub-filters used to model chromatic effects within a band.
+
+    Returns
+    -------
+    `lsst.afw.geom.Extent2I`
+        The 2D shift due to DCR, in pixels.
+    """
+    rotation = calculateRotationAngle(visitInfo, wcs)
+    dcrShift = []
+    lambdaEff = filterInfo.getFilterProperty().getLambdaEff()
+    for wl0, wl1 in wavelengthGenerator(filterInfo, dcrNumSubfilters):
+        # Note that diffRefractAmp can be negative, since it's relative to the midpoint of the full band
+        diffRefractAmp0 = differentialRefraction(wl0, lambdaEff,
+                                                 elevation=visitInfo.getBoresightAzAlt().getLatitude(),
+                                                 observatory=visitInfo.getObservatory(),
+                                                 weather=visitInfo.getWeather())
+        diffRefractAmp1 = differentialRefraction(wl1, lambdaEff,
+                                                 elevation=visitInfo.getBoresightAzAlt().getLatitude(),
+                                                 observatory=visitInfo.getObservatory(),
+                                                 weather=visitInfo.getWeather())
+        diffRefractAmp = (diffRefractAmp0 + diffRefractAmp1)/2.
+        diffRefractPix = diffRefractAmp.asArcseconds()/wcs.getPixelScale().asArcseconds()
+        dcrShift.append(afwGeom.Extent2D(diffRefractPix*np.cos(rotation.asRadians()),
+                                         diffRefractPix*np.sin(rotation.asRadians())))
+    return dcrShift
+
+
+def calculateRotationAngle(visitInfo, wcs):
```

Comment:
I think this should be named `calculateImageParallacticAngle` or similar. The name "parallactic" needs to be in there.